### PR TITLE
docs: Adding blog submission guidelines

### PR DESCRIPTION
**Pull request by ildikov**
The Kata blog was recently moved to the project's website. The content of the blog is stored together with the rest of the website source on GitHub.

This patch adds a short guide that describes how to submit a new blog post as a PR, to appear on the project's website.

### CODE REVIEW

File: docs/Blog-Post-Submission-Guide.md
Comment (stevenhorsman) [inline docs/Blog-Post-Submission-Guide.md:6]:
```suggestion
## Share your stories on the Kata Containers Website
```

File: docs/Blog-Post-Submission-Guide.md
Comment (stevenhorsman) [inline docs/Blog-Post-Submission-Guide.md:53]: Should this be in triple backticks so it shows up as code properly?

File: docs/Blog-Post-Submission-Guide.md
Comment (stevenhorsman) [inline docs/Blog-Post-Submission-Guide.md:43]: I guess this is a non GFM way of showing code, so it's required here?

File: docs/Blog-Post-Submission-Guide.md
Comment (stevenhorsman) [inline docs/Blog-Post-Submission-Guide.md:77]: Same comment as above - triple backticks needed here.

**Review by stevenhorsman — COMMENTED** [review]
A few updates needed and the commit header needs to start with `docs:` to match the patch format:
```
subsystem: One line change summary

More detailed explanation of your changes (why and how)
that spans as many lines as required.

Signed-off-by: Contributors Name <contributor@foo.com>
```
Otherwise it looks good. Thanks!

File: docs/Blog-Post-Submission-Guide.md
Comment (ildikov) [reply docs/Blog-Post-Submission-Guide.md:6]: I should've used the editor rather than my eyes to double check there's no copy-paste error! Good catch!

File: docs/Blog-Post-Submission-Guide.md
Comment (sprt) [inline docs/Blog-Post-Submission-Guide.md:53]: Going even further, should we create a template blog post .md file that contains this metadata and shows how to include an image, as described below?

Comment (ildikov) [reply docs/Blog-Post-Submission-Guide.md:53]: @sprt Thank you for bringing up the idea!

I think this would potentially be more work than how useful it is. If the template file is in the website repo, it'll get rendered to the blog, unless it is specifically ignored. If we have the template file here, then people will need to download it from here to use it. At that point, I think it is easier to have them copy text from here, or an already existing blog post. I often do that myself.

The StarlingX community's blog works the same way as the new Kata blog, and uses this guide. People so far have been submitting blog posts without any issues, or questions.

Comment (ildikov) [reply docs/Blog-Post-Submission-Guide.md:52]: Fixed now.

**Review by stevenhorsman — COMMENTED** [review]
Hi @ildikov - the changes look good to me. In order to comply with the kata-containers commit check can you fixup up the commits into a single one and add `docs: ` to the commit header. Thanks!

**Review by stevenhorsman — APPROVED** [review]
LGTM. Thanks @ildikov!

**Review by danmihai1 — APPROVED** [review]

**Comment by stevenhorsman** [comment]
The static-check spellcheck failed with a few entries:
```
2024-05-30T14:58:09.7497895Z [kata-spell-check.sh:125] INFO: Spell checking file 'docs/Blog-Post-Submission-Guide.md'
2024-05-30T14:58:09.9518936Z [kata-spell-check.sh:275] WARNING: Word ''': did you mean one of the following?: e, s, i, a, n, r, t, o, l, c, d, u, g, m, p
2024-05-30T14:58:09.9551457Z [kata-spell-check.sh:275] WARNING: Word ''': did you mean one of the following?: e, s, i, a, n, r, t, o, l, c, d, u, g, m, p
2024-05-30T14:58:09.9584848Z [kata-spell-check.sh:275] WARNING: Word ''Annual': did you mean one of the following?: annual
2024-05-30T14:58:09.9618166Z [kata-spell-check.sh:275] WARNING: Word 'Report'': did you mean one of the following?: Report, Report's, Reports
2024-05-30T14:58:09.9652018Z [kata-spell-check.sh:275] WARNING: Word 'communitys': did you mean one of the following?: community, community's, community s, communists, communist, communism
2024-05-30T14:58:09.9685340Z [kata-spell-check.sh:275] WARNING: Word 'img'': did you mean one of the following?: image
2024-05-30T14:58:09.9718223Z [kata-spell-check.sh:275] WARNING: Word 'md'': did you mean one of the following?: MD
2024-05-30T14:58:09.9753749Z [kata-spell-check.sh:275] WARNING: Word 'src': did you mean one of the following?: arc, sec, sic, sac, soc, orc, SRO
2024-05-30T14:58:09.9761895Z [kata-spell-check.sh:284] ERROR: Spell check failed for file: 'docs/Blog-Post-Submission-Guide.md'
```
some of which we can fix, but others might need things put in backticks, or our dictionary files updated to get through.

**Comment by ildikov** [comment]
@stevenhorsman Thank you for pointing those out. When it comes to the ones that we cannot fix, what's the preference in backticks vs updating dictionary files?

**Comment by stevenhorsman** [comment]
> @stevenhorsman Thank you for pointing those out. When it comes to the ones that we cannot fix, what's the preference in backticks vs updating dictionary files?

It depends on the context. Things like `'static/img'`, `'.md'` and `’src/pages/blog/'` are easy to switch from `'` to backticks and probably better. 
For `'Annual Report'` it's not as natural a fix, but is probably the path of least resistance to put it in backticks rather than work out how to get the spell checker to be happy with single quotes, but we might be able to resolve that with some effort.

**Comment by ildikov** [comment]
It's also complaining about "communitys", but there's only "community's" in the actual text. Do I need to put ' in backticks or how does that work?

**Comment by ildikov** [comment]
> It's also complaining about "communitys", but there's only "community's" in the actual text. Do I need to put ' in backticks or how does that work?

Never mind, I think I found the issue, it was somehow the wrong kind of ', even though I can't seem to find how I was able to put that there in the first place...

**Comment by stevenhorsman** [comment]
Only the metric tests are required an failing now. We have a wider issues with these, so we might just need to wait until this is resolved before getting this merged. Hopefully it should be soon.

**Comment by ildikov** [comment]
@stevenhorsman Just to confirm, those tests are unrelated to this change, right?

**Comment by stevenhorsman** [comment]
> @stevenhorsman Just to confirm, those tests are unrelated to this change, right?

Yes, I've tried to clean up the server they run on and re-running them now, so hopefully we'll get them green soon.

**Comment by ildikov** [comment]
Sounds good, thank you!

For future PRs, is there a way to not trigger non-relevant tests when something is clearly a documentation change?

**Comment by stevenhorsman** [comment]
> Sounds good, thank you!
> 
> For future PRs, is there a way to not trigger non-relevant tests when something is clearly a documentation change?

Yes - that is being worked on as part of the changes that Lukas demo'd on the AC meeting a few weeks ago.

**Pull request merged**
